### PR TITLE
feat(Alert): Add new `clickableTextProps` prop

### LINKED ISSUE #334: Alert anchor has no ability to pass href instead of onclick
Now we expect onClick handler for clickable text. We put anchor there, so would be better to have href for most of the cases.
We might need to rethink that API for v2 even..

### PR DESCRIPTION
**Pull request by mayank99**
Deprecated `onClick` and added `clickableTextProps` to allow passing href.

Updated class names using iTwinUI-css 0.34.0 (from https://github.com/iTwin/iTwinUI/pull/318).

Closes #334

## Checklist

- [x] Add meaningful unit tests for your component (verify that all lines are covered)
- [x] Verify that all existing tests pass
- [ ] Add component features demo in Storybook (different stories)
- [ ] ~~Approve test images for new stories~~
- [ ] ~~Add screenshots of the key elements of the component~~

### CODE REVIEW

File: src/core/Alert/Alert.tsx [hunk 1]
```diff
@@ -75,11 +80,15 @@ export const Alert = (props: AlertProps) => {
...
     >
-      {<StatusIcon className='iui-icon' />}
-      <span className='iui-message'>
+      <StatusIcon className='iui-alert-icon' />
+      <span className='iui-alert-message'>
         {children}
         {clickableText && (
-          <a style={{ cursor: 'pointer' }} onClick={onClick}>
+          <a
+            className='iui-alert-link'
```
Comment (veekeys) [inline src/core/Alert/Alert.tsx:88]: where is merging?

File: src/core/Alert/Alert.tsx [hunk 2]
```diff
@@ -20,7 +20,11 @@ export type AlertProps = {
    */
   clickableText?: React.ReactNode;
   /**
-   * Action handler for the clickable text.
+   * Props for the clickable text. Used for providing `href` and other attributes.
+   */
+  clickableTextProps?: React.ComponentPropsWithoutRef<'a'>;
+  /**
+   * Action handler for the clickable text, used when href is not provided.
```
Comment (veekeys) [inline src/core/Alert/Alert.tsx:27]: I dont think adding comment `when href is not provided` is actually right :D

File: src/core/Alert/Alert.tsx [hunk 3]
```diff
@@ -20,7 +20,11 @@ export type AlertProps = {
    */
   clickableText?: React.ReactNode;
   /**
-   * Action handler for the clickable text.
+   * Props for the clickable text. Used for providing `href` and other attributes.
+   */
+  clickableTextProps?: React.ComponentPropsWithoutRef<'a'>;
```
[the remark this answers is not in the document]
Comment (mayank99) [reply src/core/Alert/Alert.tsx:25]: ok changed, but not sure it will work

File: src/core/Alert/Alert.tsx (hunk 2 above)
Comment (mayank99) [reply src/core/Alert/Alert.tsx]: Added deprecation

**Review by veekeys — APPROVED** [review]

**Pull request merged**